impl/src/main/java/org/jboss/weld/annotated/enhanced/jlr/EnhancedAnnotatedTypeImpl.java

```diff
@@ -158,7 +144,18 @@ protected EnhancedAnnotatedTypeImpl(SlimAnnotatedType<T> annotatedType,
                 this.superclass = classTransformer.getEnhancedAnnotatedType(superclass, slim.getIdentifier().getBdaId());
             }
         } else {
-            this.superclass = classTransformer.getEnhancedAnnotatedType(Object.class, AnnotatedTypeIdentifier.NULL_BDA_ID);
+            // TODO do we need the try-catch block? Since the class comes from an extension, do we always have access to it superclasses?
```

Comment:
I am really not sure about this - we seemed to have been very deliberate in using `Object` as superclass here